tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -14,33 +14,90 @@ else
 }
 
 
-function Get-AzProfile 
+function Get-ProfileCachePath
 {
-  @{
-    '2015-05' = 
-    @{
-        'AzureRM' = @('1.2.4');
-        'AzureRM.Storage' = @('1.2.4');
-        'Azure.Storage' = @('1.2.4');
-        'AzureRM.Profile' = @('1.2.4');
-     }; 
-     '2016-09' = 
-     @{
-        'AzureRM' = @('2.0.1');
-        'AzureRM.Storage' = @('2.0.1')
-        'Azure.Storage' = @('2.0.1');
-        'AzureRM.Profile' = @('2.0.1');
-     }
-   }
+  $ProfileCache = Join-Path -path $env:LOCALAPPDATA -childpath "Microsoft\AzurePowerShell\ProfileCache"
+  return $ProfileCache
+}
+
+
+# Get-ProfileMap from Azure Endpoint
+function Get-AzureProfileMap
+{
+  $ProfileCache = Get-ProfileCachePath
```

Comment:
We need to have a default map that ships with the cmdlets, so that we can work if this call fails.  We probably also want to have an aging mechanism for the cache, so that we don't download a new copy of the map on each request.